datatypes/src/operations/image/to_png.rs

```diff
@@ -16,15 +18,18 @@ where
     fn to_png(&self, width: u32, height: u32, colorizer: &Colorizer) -> Result<Vec<u8>> {
         // TODO: use PNG color palette once it is available
 
-        let (.., raster_y_size, raster_x_size) = self.dimension().as_pattern();
+        let [.., raster_y_size, raster_x_size] =
+            self.grid_dimension.size_as_index().as_index_array();
         let scale_x = (raster_x_size as f64) / f64::from(width);
         let scale_y = (raster_y_size as f64) / f64::from(height);
 
         let color_mapper = colorizer.create_color_mapper();
 
         let image_buffer: RgbaImage = RgbaImage::from_fn(width, height, |x, y| {
             let (grid_pixel_x, grid_pixel_y) = image_pixel_to_raster_pixel(x, y, scale_x, scale_y);
-            if let Ok(pixel_value) = self.pixel_value_at_grid_index(&(grid_pixel_y, grid_pixel_x)) {
+            if let Ok(pixel_value) =
+                self.pixel_value_at_grid_index(&[grid_pixel_y, grid_pixel_x].into())
```

Comment:
didnt you want to aliminate the `into`s?

Reply:
yes but it is hard enough to get this working as it is.

Reply:
removed the intos